web/page0/src/app/host.service.ts

```diff
@@ -1,30 +1,102 @@
 import {Injectable} from '@angular/core';
-import {hostListSelector} from 'src/app/store/selectors';
 import {Store} from 'src/app/store/store';
 import {ApiService} from './api.service';
 import {HostInstance} from 'src/app/interface/cloud-orchestrator.dto';
 import {Runtime} from 'src/app/interface/runtime-interface';
+import {throwError} from 'rxjs';
+import {catchError, tap} from 'rxjs/operators';
+import {OperationService} from './operation.service';
+import {HostStatus} from './interface/host-interface';
 
 @Injectable({
   providedIn: 'root',
 })
 export class HostService {
   createHost(hostInstance: HostInstance, runtime: Runtime, zone: string) {
-    // TODO: long polling
-    return this.apiService.createHost(runtime.url, zone, {
-      host_instance: hostInstance,
-    });
+    return this.apiService
+      .createHost(runtime.url, zone, {
+        host_instance: hostInstance,
+      })
+      .pipe(
+        tap(operation => {
+          const waitUrl = `${runtime.url}/v1/zones/${zone}/operations/${operation.name}`;
+          this.store.dispatch({
+            type: 'host-create-start',
+            wait: {
+              waitUrl,
+              metadata: {
+                type: 'host-create',
+                zone,
+                runtimeAlias: runtime.alias,
+              },
+            },
+          });
+          this.waitService.longPolling<HostInstance>(waitUrl).subscribe({
```

Comment:
instead of subscribe in tap, you can use switch map for that instead.
tap(operation => this.store.dispatch({...}))
.switchMap(operation => this.waitService.longPolling~~~)?

Reply:
Subscription from tap is needed to meet current requirements:
- On completion of POST /hosts, the user navigates back to previousUrl immediately
- EnvService does not have dependencies on RouterModule or Snackbar as they are sort of UI modules